### ISSUE TITLE
New Firely validator does not complain when extension is used outside of its context

### ISSUE THREAD

**Issue by wardweistra** [FirelyTeam/firely-net-sdk#2643]
When an extension is used in the root of a Location resource: <https://simplifier.net/snippet/wardweistra/66>
But the extension has as context defined the element `Address`: <https://simplifier.net/Signal-MSO-FHIR-Profiles/SignalRegion/~xml>
We do not yet throw an error in the new validator.

[Slack Message](https://firelyteam.slack.com/archives/C011XD1TKL0/p1702898118334729?thread_ts=1702898118.334729&cid=C011XD1TKL0)

**Comment by wardweistra** [comment]
![image](https://github.com/FirelyTeam/firely-net-sdk/assets/779157/31a9513b-f8d0-4aaa-83f6-172eeaf11eb2)
Confirmed with the [Firely Validation Demo](https://github.com/FirelyTeam/Firely.Fhir.ValidationDemo) that it passes validation with warnings/errors.

**Comment by ewoutkramer** [comment]
I can confirm this blindly: We don't check the context at all - never implemented that!

**Comment by ewoutkramer** [comment]
BTW: We're trying to avoid having validator issues here. We will need to transfer this issue to the validator repo, but since this is a private repo (at this moment), I cannot use GitHub's transfer funtion yet, so have to do that when we open up the repo in january!

**Comment by mikaelweave** [comment]
@ewoutkramer - do you know if the old validator checked extension context? I'm working with a support issue on this very profile and am making sure I understand what's happening. Thank you! 😄

**Comment by alexzautke** [comment]
No, extension context checking was never implemented in the old validator.

**Comment by Narnian12** [comment]
@ewoutkramer do you happen to know when the new validator will be public? We want to start using profile validation within our code, and have been trying to use the legacy one, but figured it would be much better to use the more recent one.

**Comment by mikaelweave** [comment]
Thanks @alexzautke, appreciate the confirmation. Looks like we need to update to the new one. 😄
